src/main/java/com/microsoft/sqlserver/jdbc/SQLServerVSMEnclaveProvider.java

```
@@ -139,33 +140,27 @@ private VSMAttestationResponse validateAttestationResponse(VSMAttestationRespons
-            if (connection.enclaveEstablished()) {
-                rs = executeSDPEv1(stmt, userSql, preparedTypeDefinitions);
-            } else {
-                rs = executeSDPEv2(stmt, userSql, preparedTypeDefinitions, vsmParams);
-            }
-            if (null == rs) {
-                // No results. Meaning no parameter.
-                // Should never happen.
-                return enclaveRequestedCEKs;
-            }
-            processSDPEv1(userSql, preparedTypeDefinitions, params, parameterNames, connection, stmt, rs,
-                    enclaveRequestedCEKs);
-            // Process the third resultset.
-            if (connection.isAEv2() && stmt.getMoreResults()) {
-                rs = (SQLServerResultSet) stmt.getResultSet();
-                while (rs.next()) {
-                    hgsResponse = new VSMAttestationResponse(rs.getBytes(1));
-                    // This validates and establishes the enclave session if valid
-                    if (!connection.enclaveEstablished()) {
-                        hgsResponse = validateAttestationResponse(hgsResponse);
+            try (ResultSet rs = connection.enclaveEstablished() ? executeSDPEv1(stmt, userSql,
+                    preparedTypeDefinitions) : executeSDPEv2(stmt, userSql, preparedTypeDefinitions, vsmParams)) {
+                if (null == rs) {
+                    // No results. Meaning no parameter.
+                    // Should never happen.
+                    return enclaveRequestedCEKs;
+                }
+                processSDPEv1(userSql, preparedTypeDefinitions, params, parameterNames, connection, stmt, rs,
+                        enclaveRequestedCEKs);
+                // Process the third resultset.
+                if (connection.isAEv2() && stmt.getMoreResults()) {
+                    try (ResultSet hgsRs = (SQLServerResultSet) stmt.getResultSet()) {
+                        if (hgsRs.next()) {
```

Is it possible that the resultset is empty here? If it is, should we throw an exception or it is a legit scenario?

---

It shouldn't happen but something can go wrong, we can throw an exception.